enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+
+
+#### Cloud Credential Operator
+
+Cloud Credential Operator is responsible for handling `CredentialsRequest` custom resource.
+`CredentialsRequest`s allow OpenShift components to request credentials for a particular cloud provider.
+
+On unsupported platforms the operator goes into no-op mode, which technically is mostly an equivalent
+of the ["Manual"](https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-manual.html)
+mode.
+
+Cloud Credential operator [uses](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/operator/controller.go#L74) an "actuator pattern"
+and, in theory, in the future it might be extended in a way to react for the new "External" platform type and allow users to supplement their own
+platform-specific credentials management logic.
+
+During initial enablement phases of "External" platform type, no specific actions will be needed there, since
+CCO [would](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/operator/controller.go#L134) go
+into no-op mode if it encounters unrecognized platform.
+
+#### Cluster Image Registry Operator
+
+For image registry a [storage backend config decision](https://github.com/openshift/cluster-image-registry-operator/blob/99474318db709a6d17d06468b90cdf0dc0fd2b87/pkg/storage/storage.go#L157) is platform specific.
+On a "None" platform type Image Registry Operator goes into no-op state. For unknown platforms the image registry is configured
+with EmptyDir storage. 
+
+Also, Image Registry storage options might be configured to use pvc backed or external storage systems
+(such as Ceph or S3 compatible object storage) as a day 2 operation.
+
+For now it seems that no particular action for the "External" platform type needed within Image Registry Operator,
+since we're providing enough possibilities to customize Image Registry storage backend.
+
+#### Cluster Ingress Operator
+
+Within the ingress operator PlatformType affects on two things:
+1. Choosing EndpointPublishingStrategy, which is `HostNetworkStrategyType` for "None" and unknown PlatformType
+2. Creating dns provider on some platforms. This logic does not engaged for "None" or unknown PlatformType
+
+Speaking about `LoadBalancer` EndpointPublishingStrategy - Cluster Ingress Operator might be reconfigured for using it
+as day2 operation. Operator itself  creates `Service` objects with correct provider-specific annotations, actual handling
+of such objects happens in a provider-specific [cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
+
+#### Cluster Network Operator
+
+TBD
+
+### Workflow Description
+
+Since proposed change intended to have no effect during initial implementation, no specific user interaction is expected.
+"External" platform type would be treated the same as "None" platform.
```

Comment:
i think we should change this a little to say that during phase 1 there will be no expectation of change for the user, but that after phase 3 is complete we expect that a user could change the platform and set the spec, as well as capabilities, to customize their openshift deployments.